cmake/extensions.cmake

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# cddl_generate(<target> <cddl_file> ENTRY_TYPES <entry_types> [VERBOSE])
+#
+# Add generated code to the project for decoding CBOR.
+#
+# <cddl_file> is a file with CDDL data describing the expected CBOR data.
+#
+# <entry_types> is a list of the types defined in the CDDL file for which there
+#               should be an exposed decoding function.
+#
+# Specify VERBOSE to add printing to the generated code. Check kconfig to add
+# printing in the low level CBOR decoding code.
+#
+# The result of the function is that a library is added to the project which
+# contains the generated decoding code. The code is generated at build time using
+# a script, and generates a c file and an h file with names derived from the
+# name of the <cddl_file>.
+function(cddl_generate target cddl_file)
+  if(IS_ABSOLUTE ${cddl_file})
+    set(cddl_path ${cddl_file})
+  else()
+    set(cddl_path ${CMAKE_CURRENT_LIST_DIR}/${cddl_file})
+  endif()
+
+  if(NOT EXISTS ${cddl_path})
+    message(FATAL_ERROR "CDDL input file ${cddl_file} does not exist.")
+  endif()
+
+  get_filename_component(name ${cddl_path} NAME_WE)
+  set(c_file ${PROJECT_BINARY_DIR}/${name}.c)
+  set(h_file_dir ${PROJECT_BINARY_DIR}/zephyr/include/generated)
+  set(h_file ${h_file_dir}/${name}.h)
+
+  cmake_parse_arguments(CDDL "VERBOSE" "" "ENTRY_TYPES" ${ARGN})
+
+  if (NOT DEFINED PYTHON_EXECUTABLE)
+    set(PYTHON_EXECUTABLE python3)
+  endif()
```

Comment:
Are you expecting this to be used outside NCS Scope ?
If not, then I suggest to remove those lines.

If this should be usable outside NCS scope, then Python should be found in proper CMake way.
Setting python to `python3` will for sure break on windows, and on Linux you have no control of which Python3 version you get.

Proper CMake way is: `find_package`
https://cmake.org/cmake/help/latest/module/FindPython.html

Note, that CMake <3.13 has a detection flaw on windows: 
https://github.com/zephyrproject-rtos/zephyr/issues/11103
and fixed in Zephyr:
https://github.com/zephyrproject-rtos/zephyr/pull/11306
https://github.com/zephyrproject-rtos/zephyr/pull/11306/files

Reply:
Hmm good point. This project will be integrated into wherever it is used, so maybe we don't need to focus on making a standalone build system, and just say that the cmake code requires Zephyr?

Reply:
and simply state that if used outside NCS (Zephyr) then it is required that PYTHON_EXECUTABLE is set.

Reply:
It also uses zephyr_interface. When I didn't have that, I got weird errors about mismatched ABIs.

Reply:
> It also uses zephyr_interface. 

yep, I noticed.
And I did consider if this should be a responsibility for the caller of `add_library` to link, but as we do have the knowledge internally, and i'm unsure which parts of Zephyr interface you need, I find it ok to link internally, for time being.
That can avoid chasing down strange bugs.